### PR TITLE
queue_close.rb - refactor loading, move comment for docs

### PR DESCRIPTION
**Pull request by MSP-Greg**
### Description

Small refactor of queue_close.rb and its loading.  Move comment so it is attached to `QueueClose`.

### Your checklist for this pull request
- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added an entry to [History.md](../blob/master/History.md) if this PR fixes a bug or adds a feature. If it doesn't need an entry to HISTORY.md, I have added `[changelog skip]` or `[ci skip]` to the pull request title.
- [ ] I have added appropriate tests if this PR fixes a bug or adds a feature.
- [ ] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [ ] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [ ] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: lib/puma/queue_close.rb [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+    def closed?
+      @closed
+    end
+    def push(object)
+      raise ClosedQueueError if @closed
+      super
+    end
+    alias << push
   end
+  ::Queue.prepend QueueClose
```
Comment (cjlarose) [inline lib/puma/queue_close.rb:24]: This comment isn't really 100% related to your changes @MSP-Greg , but I'm wondering if it strikes you as weird that Puma essentially monkey patches a class provided by the stdlib.

It seems like it'd be safer for us to just subclass Queue into a new puma-specific Queue class that provided the appropriate overrides for `close`, `closed?`, and `push`.

Comment (MSP-Greg) [reply lib/puma/queue_close.rb:24]: I believe this is only used on Ruby 2.2?

A while ago I suggested continuing support of 2.2 & 2.3, but for Puma 6.0, maybe it's time...

Comment (cjlarose) [reply lib/puma/queue_close.rb:24]: Oh yeah that makes sense. Might as well keep the monkey patch in the code base until we remove Ruby 2.2 support. 

Ruby 2.2 was EOL'd on March 31, 2018. Ruby 2.3 was EOL'd on March 31, 2019.

**Comment by MSP-Greg** [comment]
@nateberkopec - 

See CI for Ruby 2.2. Lots of `warning: instance variable @closed not initialized` shown.  I can push a commit as below?  Just checked in my fork...

from:
```ruby
    def push(object)
      raise ClosedQueueError if @closed
      super
    end
```
to, adding line `@closed ||= false`:
```ruby
    def push(object)
      @closed ||= false
      raise ClosedQueueError if @closed
      super
    end
```

**Comment by nateberkopec** [comment]
@MSP-Greg

**Comment by MSP-Greg** [comment]
Done

**Pull request merged**